software/database/src/main/java/org/apache/brooklyn/entity/database/mysql/MySqlNodeImpl.java

```diff
@@ -135,8 +135,8 @@ protected void disconnectSensors() {
     public int getPort() {
         return getAttribute(MYSQL_PORT);
     }
-    
-    public String getSocketUid() {
+
+    public synchronized String getSocketUid() {
```

Comment:
Post merge comment... I'd have preferred if we'd declared a `private final Object mutex` to synchronize on. This is probably ok, but in general I don't like synchronizing on `this` in case other threads/callers have synchronized on this in other situations. No strong feelings.